lib/cisco_node_utils/vlan.rb

```diff
@@ -199,5 +248,88 @@ def mapped_vni=(vni)
     def default_mapped_vni
       config_get_default('vlan', 'mapped_vni')
     end
+
+    def private_vlan_type
+      config_get('vlan', 'private_vlan_type', id: @vlan_id)
+    end
+
+    def private_vlan_type=(pv_type)
+      Feature.private_vlan_enable
+      fail TypeError unless pv_type && pv_type.is_a?(String)
+      set_args_keys_default
+      if pv_type == default_private_vlan_type
+        pv_type = private_vlan_type
+        @set_args[:vlan] = @vlan_id
+        @set_args[:state] = 'no'
+        @set_args[:type] = pv_type
+        ignore_msg = 'Warning: Private-VLAN CLI removed'
+      else
+        @set_args[:vlan] = @vlan_id
+        @set_args[:state] = ''
+        @set_args[:type] = pv_type
+        ignore_msg = 'Warning: Private-VLAN CLI entered'
+      end
+      result = config_set('vlan', 'private_vlan_type', @set_args)
+      cli_error_check(result, ignore_msg)
+    end
+
+    def default_private_vlan_type
+      config_get_default('vlan', 'private_vlan_type')
+    end
+
+    def private_vlan_association
+      result = config_get('vlan', 'private_vlan_association',
+                          id: @vlan_id)
+      result.gsub(/["\[\]" "]/, '')
+    end
+
+    def private_vlan_association=(config)
+      fail TypeError unless config &&
+                            config.is_a?(String)
+      Feature.private_vlan_enable
+      set_args_keys_default
+      if config == default_private_vlan_association
+        @set_args[:vlan] = @vlan_id
+        @set_args[:state] = 'no'
+        @set_args[:vlans] = config
+        result = config_set('vlan', 'private_vlan_association',
+                            @set_args)
+        cli_error_check(result)
+      else
+        vlan_list = private_vlan_association
+        if vlan_list.empty?
+          @set_args[:vlan] = @vlan_id
+          @set_args[:state] = ''
+          @set_args[:vlans] = config
+          result = config_set('vlan', 'private_vlan_association',
+                              @set_args)
+          cli_error_check(result)
+        else
+          compute_vlan_list_association(vlan_list, config)
+          vlan_to_remove = @set_args[:vlan_to_remove]
+          vlan_to_config = @set_args[:vlan_to_config]
+          set_args_keys_default
+          unless vlan_to_remove.empty?
+            @set_args[:vlan] = @vlan_id
+            @set_args[:state] = 'no'
+            @set_args[:vlans] = vlan_to_remove
+            result = config_set('vlan', 'private_vlan_association',
+                                @set_args)
+            cli_error_check(result)
+
+            @set_args[:vlan] = @vlan_id
+            @set_args[:state] = ''
+            @set_args[:vlans] = vlan_to_config
+            result = config_set('vlan', 'private_vlan_association',
+                                @set_args)
+            cli_error_check(result)
```

Comment:
Instead of 4 calls to `cli_error_check(result)` can you just have one at the end of the method outside the if/else block?